### ISSUE TITLE
JWS has invalid anti-replay nonce

### ISSUE THREAD

**Issue by jfisbein** [xenolf/lego#366]
After the closing of #339 I tried to generate the certificates but I still get the same error:

This are the commands I runned:
(changed some data for privacy)
```
$ date
Mon Mar 20 21:15:53 CET 2017

$ mkdir -p /root/lego
$ docker pull xenolf/lego
Using default tag: latest
latest: Pulling from xenolf/lego

709515475419: Already exists 
08590bbbf3a6: Pull complete 
Digest: sha256:81eb513d461b6a2ed03b055fddcc33cb906b21b8c6a4d33b3f3f483952175edc
Status: Downloaded newer image for xenolf/lego:latest

$ docker run --rm --name lego -v /root/lego:/.lego --env GANDI_API_KEY=xxxxxxxxxxxxxxxxxxx xenolf/lego --domains joan.domain.com --email joan@domain.com --accept-tos --dns gandi --dns-resolvers a.dns.gandi.net --pem run
2017/03/20 20:16:12 [INFO][joan.domain.com] acme: Obtaining bundled SAN certificate
2017/03/20 20:16:12 [INFO][joan.domain.com] AuthURL: https://acme-v01.api.letsencrypt.org/acme/authz/ESZNvYshFPRkWSybn9Huo4QqVfR4lvLIpQxeZ_cTWZQ
2017/03/20 20:16:12 [INFO][joan.domain.com] acme: Trying to solve DNS-01
2017/03/20 20:16:15 [INFO][joan.domain.com] Checking DNS record propagation using [a.dns.gandi.net:53]
2017/03/20 20:34:23 [joan.domain.com] Could not obtain certificates
	acme: Error 400 - urn:acme:error:badNonce - JWS has invalid anti-replay nonce BOyYLPtyguoonH3VnVm1pr0NIBd6d5g6mbztjU2BX28
```

Does the latest docker image includes the changes?

**Comment by xenolf** [comment]
@jfisbein please try again with current master. I placed the fix on a code path where it was never hit, #367 fixes this.

**Comment by jfisbein** [comment]
ir worked!!
thanks!

```
2017/03/30 08:04:28 [INFO][joan.mydomain.com] acme: Obtaining bundled SAN certificate
2017/03/30 08:04:28 [INFO][joan.mydomain.com] AuthURL: https://acme-v01.api.letsencrypt.org/acme/authz/nIDbvAmqBn7_9se6DAXq5FIa3_wXFNLOejLQx5MGz18
2017/03/30 08:04:28 [INFO][joan.mydomain.com] acme: Could not find solver for: http-01
2017/03/30 08:04:28 [INFO][joan.mydomain.com] acme: Trying to solve DNS-01
2017/03/30 08:04:31 [INFO][joan.mydomain.com] Checking DNS record propagation using [a.dns.gandi.net:53]
2017/03/30 08:31:55 [INFO][joan.mydomain.com] The server validated our request
2017/03/30 08:31:56 [INFO][joan.mydomain.com] acme: Validations succeeded; requesting certificates
2017/03/30 08:31:57 [INFO] acme: Requesting issuer cert from https://acme-v01.api.letsencrypt.org/acme/issuer-cert
2017/03/30 08:31:57 [INFO][joan.mydomain.com] Server responded with a certificate.
```
